proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+suffix (unless specifically excluded using the `excludePrivateSuffixes` option).
+
+##### Example
+
+`domain` = foo.bar.lib.de.us
+
+| Candidate Suffix | PSL Section |
+|-----------------:|:-----------:|
+|            de.us | ICANN       |
+|        lib.de.us | Private     |
+
+The longest is lib.de.us, so `getRegistrableDomain()` resolves to bar.lib.de.us
+
+#### 3. PSL Special Rules
+
+The lookup performed by `getRegistrableDomain()` should adhere to the
+[PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm).
+In particular, it should apply the 'wildcard' and 'exception' rules in the PSL.
+
+##### Examples
+
+| Domain                   | Public suffix | Matched PSL rule | Explanation    | Registrable Domain |
+|-------------------------:|--------------:|-----------------:|:--------------:|-------------------:|
+| sub.domain.com           | com           | com              | Simple rule    | domain.com         |
+| sub.domain.co.uk         | co.uk         | co.uk            | Simple rule    | domain.co.uk       |
+| sub.domain.gov.ck        | gov.ck        | *.ck             | Wildcard rule  | domain.gov.ck      |
+| sub.domain.any.ck        | any.ck        | *.ck             | Wildcard rule  | domain.any.ck      |
+| sub.sub.domain.any.ck    | any.ck        | *.ck             | Wildcard rule  | domain.any.ck      |
+| www.ck                   | ck            | !www.ck          | Exception rule | www.ck             |
+| sub.www.ck               | ck            | !www.ck          | Exception rule | www.ck             |
+| sub.sub.www.ck           | ck            | !www.ck          | Exception rule | www.ck             |
+
+#### 4. Unknown Suffixes
+
+If no matching suffix is found in the PSL for a `domain` parameter, then unless it is determined
+to be specifically [invalid](#6-invalid-domain-parameter), it should be assumed the domain has a
+single-label suffix.
+
+##### Example
+
+| Domain parameter       | Registrable domain | PSL Section |
+|-----------------------:|-------------------:|:-----------:|
+| www.example.foobar     | example.foobar     | n/a         |
+| www.example.co.foobar  | co.foobar          | n/a         |
+
+#### 5. IDN
+
+The `domain` parameter passed to `getRegistrableDomain()` may be either Unicode
+or Punycode.
+
+When settling the promise returned by `getRegistrableDomain()`, the resulting
+domain name should be converted to Unicode from Punycode by default.
+
+However, if an `options` object is passed to `getRegistrableDomain()` with key
+`punycode` set to `true`, then Punycode should be used instead.
+
+##### Example
+
+`domain` = foo.bar.example.مليسيا
+
+| Option                     |     Registrable Domain |
+|----------------------------|-----------------------:|
+| punycode = false (default) |         example.مليسيا |
+| punycode = true            | example.xn--mgbx4cd0ab |
+
+#### 6. Invalid domain parameter
```

Comment:
This is important to keep in mind.`isSameSite("100.200.30.2", "100.200.31.2")` should probably return false so even if the last label is assumed to be a TLD (and thus not checked) there should be an exception that checks for IPv4 addresses.